### DISCUSSION TITLE
an auto build all-in-one bundle.

### DISCUSSION THREAD

**Discussion by Readon** [SpinalHDL/SpinalHDL#878] [Ideas]
If there is an all-in-one bundle like oss-cad-build (it has too much but sbt and jre), then it would help people work in company with poor/no internet connection and starter who has difficulties on all installation process.

**Comment by andreasWallner** [comment]
No package to just unzip like with oss-cad is currently available. In the last discussion about the the conclusion was that it seems currently there are two possible ways to use SpinalHDL in environment w/o internet:
1) Use the docker container that is available, that should have a sbt package cache with everything that is need
2) Build once on a machine with internet connection and copy the resulting cache.
At work I use the artifactory mirror that we have to fetch through it, but it was quite annoying to set sbt up to use it...

**Reply by Readon** [reply]
The problem of docker container is that it has to be auto build and update.
Furthermore, if the docker container is necessary, then how to combine with IDE should be considered.

If we can reuse the existing oss-cad-build script and extend it to sbt, then it can be build automatically by github's facility.

**Reply by Dolu1990** [reply]
There is some utilites to handle SBT cache for offline machines : 
https://github.com/SpinalHDL/SpinalNomad/tree/main/sbt

**Reply by Readon** [reply]
That's greate.

**Reply by Dolu1990** [reply]
If you think it is kind of a good solution, it can be pushed forward. Releases of it could also be made
